### PR TITLE
Bwuni fixes

### PR DESCRIPTION
**Pull request by tcojean**
Fix two issues with tests on bwuni

The issues seem to be with the hwloc-related tests, esp. with the case where we only have access to a part of the machine.

### CODE REVIEW

**Review by upsj — APPROVED** [review]
LGTM! Thanks for fixing this, it also caused failures on my WSL instance.

**Review by pratikvn — APPROVED** [review]
LGTM!

File: cuda/test/base/cuda_executor_topology.cu [hunk 1]
```diff
@@ -149,9 +152,12 @@ TEST_F(CudaExecutor, KnowsItsNuma)
 {
     cuda = gko::CudaExecutor::create(0, gko::OmpExecutor::create());
     auto numa0 = cuda->get_closest_numa();
-    auto close_cpu0 = cuda->get_closest_pus()[0];
+    auto close_cpus0 = cuda->get_closest_pus();
```
Comment (thoasm) [inline cuda/test/base/cuda_executor_topology.cu:155]: nit: naming does no longer fit IMO
```suggestion
    auto close_cpus = cuda->get_closest_pus();
```

**Review by thoasm — APPROVED** [review]
LGTM!
This also fixes the previously failing test `cuda/test/base/cuda_executor_topology` on my machine.

**Pull request merged**